test/unit/history/then.test.js

```diff
@@ -0,0 +1,40 @@
+import History from '../../../src/history'
+
+describe('History::then', function () {
+  const action = n => n
+
+  it('allows direct interop with promises', function () {
+    const history = new History()
+
+    let one = history.append(action)
+    let two = history.append(action)
+    let three = history.append(action)
+
+    setTimeout(function() {
+      one.resolve()
+      two.resolve()
+      three.resolve()
+    }, 10)
+
+    return history
```

Comment:
Ah this could be clearer. I'm expecting that history resolves correctly. I should add a comment.